--- main.js	2020-12-19 12:10:48.670666003 +0100
+++ build/geogebra6/app/main.js	2020-12-19 12:11:15.083181535 +0100
@@ -394,10 +394,10 @@
 // if (!(process.arch === 'arm')) {
 // This code has been moved into the startup script on Raspberry Pi to allow
 // detection of Raspberry Pi version (because version 4 already supports 3D well enough).
-if (forceGpu) {
+//if (forceGpu) {
     printDebug("Ignoring GPU blacklist to enable 3D");
     app.commandLine.appendSwitch("ignore-gpu-blacklist");
-}
+//}
 
 if (process.platform === 'darwin') {
     const {systemPreferences} = require('electron');
