=== modified file 'src/i830_dri.c'
--- old/src/i830_dri.c	2008-10-09 17:55:28 +0000
+++ new/src/i830_dri.c	2009-04-14 21:08:23 +0000
@@ -1456,6 +1456,15 @@ I830DRITransitionTo2d(ScreenPtr pScreen)
 }
 
 #if DRI_SUPPORTS_CLIP_NOTIFY
+#undef planeA_x
+#undef planeA_y
+#undef planeA_w
+#undef planeA_h
+#undef planeB_x
+#undef planeB_y
+#undef planeB_w
+#undef planeB_h
+
 static void
 I830DRIClipNotify(ScreenPtr pScreen, WindowPtr *ppWin, int num)
 {

