diff -Nurp patched/src/i830_memory.c working/src/i830_memory.c
--- patched/src/i830_memory.c	2009-03-17 11:13:53.000000000 -0700
+++ working/src/i830_memory.c	2009-03-17 11:34:18.000000000 -0700
@@ -536,7 +536,7 @@ i830_allocator_init(ScrnInfoPtr pScrn, u
 		struct drm_i915_gem_init init;
 		int ret;
 
-		sp.param = I915_SETPARAM_NUM_USED_FENCES;
+		sp.param = 4;
 		if (pI830->use_drm_mode)
 		    sp.value = 0; /* kernel gets them all */
 		else if (pI830->directRenderingType == DRI_XF86DRI)
