diff -Nurp patched/src/i830_video.c working/src/i830_video.c
--- patched/src/i830_video.c	2009-04-06 16:36:56.000000000 -0700
+++ working/src/i830_video.c	2009-04-06 16:38:50.000000000 -0700
@@ -2416,6 +2416,7 @@ I830PutImage(ScrnInfoPtr pScrn,
 	    return BadAlloc;
 	if (!pPriv->textured && drm_intel_bo_pin(pPriv->buf, 4096) != 0) {
 	    drm_intel_bo_unreference(pPriv->buf);
+            pPriv->buf = NULL;
 	    xf86DrvMsg(pScrn->scrnIndex, X_ERROR,
 		       "Failed to pin xv buffer\n");
 	    return BadAlloc;
